sinks/cortex/cortex.go

```diff
@@ -207,66 +208,32 @@ func (s *CortexMetricSink) Flush(ctx context.Context, metrics []samplers.InterMe
 		return sinks.MetricFlushResult{}, nil
 	}
 
-	if s.batchWriteSize == 0 || len(metrics) <= s.batchWriteSize {
-		err := s.writeMetrics(ctx, metrics)
-		if err == nil {
-			flushedMetrics = len(metrics)
-		} else {
-			s.logger.Error(err)
-			droppedMetrics = len(metrics)
-		}
-
-		return sinks.MetricFlushResult{MetricsFlushed: flushedMetrics, MetricsDropped: droppedMetrics}, err
+	var allErrs *multierror.Error
+	batchSize := s.batchWriteSize
+	if s.batchWriteSize == 0 {
+		batchSize = len(metrics)
 	}
-
-	doIfNotDone := func(fn func() error) error {
+batching:
+	for i := 0; i < len(metrics); i += batchSize {
+		end := i + batchSize
+		if end > len(metrics) {
+			end = len(metrics)
+		}
+		batch := metrics[i:end]
```

Comment:
Previously, this made a new array&slice for batch.
Now it uses a slice from metrics.

I had to read https://go.dev/blog/slices-intro, but it looks like
1. this will be more performant
2. It's possible that changes (including append()) inside of anything that `batch` is passed to may write into the array `metrics`

I _think_ this fixes it (alternatively we can read the code for where its passed, but then we have to hope it never gets changed)?
```suggestion
		batch := metrics[i:end:end-i]
```